### PR TITLE
Feature bracketed paste on terminal

### PR DESCRIPTION
**Pull request by erw7**
Ref #11418.

### CODE REVIEW

**Comment by mckelly2833** [comment]
After updating a few lines of a merge conflict in src/nvim/option.c this works great.
Thank you @erw7 
Someone please merge when possible :-)

**Comment by erw7** [comment]
I've fixed the conflicts. For my part, I'd like to hear the opinions of other committers on the validity of the options, etc. before merging this one.

File: runtime/doc/options.txt
Comment (jamessan) [inline runtime/doc/options.txt:6284]: Why do we need this option?

Comment (erw7) [reply runtime/doc/options.txt:6284]: This is to prevent problems such as `\x1b[201~` being included in the text to be pasted, which would cause the parenthesized paste mode to complete. As far as I know, `xterm` and `mintty` have such an option. Also, although I couldn't find the option in the manual, `tmux` also converts `\x1b` to `^[` when pasting.  

#### man xterm

```
disallowedPasteControls (class DisallowedPasteControls)
               The allowPasteControls resource is normally used to prevent pasting C1 controls, as well as non-formatting C0 controls such as the ASCII
               escape character.  Those characters are simply ignored.  This resource further extends the set of control characters which cannot be pasted,
               converting each into a space.
```

#### man mintty

```
Filtering pasted text (FilterPasteControls=)
         With this setting, pasted text can be filtered for selected control characters which are then replaced by space. Filtering is not applied when
         Control is held while pasting. The option is a comma-separated list of character tags:
```

**Comment by mckellygit** [comment]
Thanks again for this great PR.
Is there anything we can do to help get it reviewed/approved/merged ?
thx,
-m

**Review by jamessan — APPROVED** [review]

**Comment by jamessan** [comment]
Assuming CI passes, this looks good to merge.

**Comment by erw7** [comment]
I'm sorry I haven't had much time lately. Changed the default for the `tpf` option, as it could be a security issue. If everything looks good, I'll merge them in a few days.

**Comment by mckellygit** [comment]
Thanks again,
-m

**Pull request merged**
